csrc/quantization/fp8/nvidia/quant_utils.cuh

```diff
@@ -508,7 +509,7 @@ __inline__ __device__ Tout convert(const Tin& x) {
   }
   #endif
   assert(false);
```

Comment:
cruft?

Reply:
@bnellnm had a similar comment on slack:
> looks like there are some redundant assert(false)'s in csrc/quantization/fp8/nvidia/quant_utils.cuh?

Pasting my reply to that here:
>I think my argument against that is that somebody might accidentally call that function and hit the __builtin_unreachable  and UB will ensue

I think that leaving the asserts in is better than deleting them. On the other hand the asserts are typically compiled away, so it's possible to hit the `__builtin_unreachable()` in release mode, so maybe we could do something better here.